contrib_bots/lib/foursquare.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import print_function
+from __future__ import absolute_import
+
+import datetime as dt
+import re
+import requests
+from os.path import expanduser
+from six.moves import configparser as cp
+from six.moves import range
+
+home = expanduser('~')
+CONFIG_PATH = home + '/zulip/contrib_bots/lib/FourSquareBot/settings.ini'
+
+def get_api_key():
+    # settings.ini must have been moved from
+    # ~/zulip/contrib_bots/lib/FourSquareBot/settings.ini into
+    # ~/settings.ini for program to work
+    # see doc.md for more information
+    with open(CONFIG_PATH) as settings:
+        config = cp.ConfigParser()
+        config.readfp(settings)
+        return config.get('FourSquare', 'api_key')
+
+class FourSquareHandler(object):
+    def __init__(self):
+        self.api_key = get_api_key()
+
+    def usage(self):
+        return '''
+            This plugin allows users to search for resturants nearby an inputted
+            location to a limit of 3 venues for every location. The name, address
+            and description of the resturant will be outputted.
+            It looks for messages starting with '@foursquare'.
+            If you need help, simply type:
+            @foursquare /help into the Compose Message box
+
+            Sample input:
+            @foursquare Chicago, IL
+            @foursquare help
+            '''
+
+    help_info = '''
+The FourSquare bot can receive keyword limiters that specify the location, distance (meters) and
+cusine of a resturant in that exact order.
+Please note the required use of quotes in the search location.
+
+Example Inputs:
+@foursquare 'Millenium Park' 8000 donuts
+@foursquare 'Melbourne, Australia' 40000 seafood
+                '''
+
+    def triage_message(self, message, client):
+        return '@FourSquare' in message['content']
+
+    def format_json(self, text, location):
+        template = ("Food nearby {location} coming right up:")
+
+        venue_list = text['response']['venues']
+        venue_details = {}
+        for i in range(len(text['response']['venues'])):
```

Comment:
Can you change this section so that you don't have to do this wordy indexing of appending i to every key?

A simpler approach like this should work:

~~~ py
        venues = text['response']['venues']

        def format_venue(venue):
            name = venue['name']
            address = venue['location']['formattedAddress']
            keyword = venue['categories'][0]['pluralName']
            blurb = '\n'.join([name, address, keyword])
            return blurb

        return '\n'.join(format_venue(venue) for venue in venues)